src/sortTemplate.ts

```diff
@@ -41,37 +41,37 @@ export class SortQuickPickItem implements vscode.QuickPickItem {
     let items: SortQuickPickItem[] = [];
-    items.push(new SortQuickPickItem("Functions", SortType.Functions, "Sort function namespaces and functions"));
-    items.push(new SortQuickPickItem("Outputs", SortType.Outputs, "Sort outputs"));
-    items.push(new SortQuickPickItem("Parameters", SortType.Parameters, "Sort parameters for the template"));
-    items.push(new SortQuickPickItem("Resources", SortType.Resources, "Sort resources based on the name including first level of child resources"));
-    items.push(new SortQuickPickItem("Variables", SortType.Variables, "Sort variables"));
-    items.push(new SortQuickPickItem("Top level", SortType.TopLevel, "Sort top level items based on recommended order (parameters, functions, variables, resources, outputs)"));
+    items.push(new SortQuickPickItem("Functions", TemplateSectionType.Functions, "Sort function namespaces and functions"));
+    items.push(new SortQuickPickItem("Outputs", TemplateSectionType.Outputs, "Sort outputs"));
+    items.push(new SortQuickPickItem("Parameters", TemplateSectionType.Parameters, "Sort parameters for the template"));
+    items.push(new SortQuickPickItem("Resources", TemplateSectionType.Resources, "Sort resources based on the name including first level of child resources"));
+    items.push(new SortQuickPickItem("Variables", TemplateSectionType.Variables, "Sort variables"));
+    items.push(new SortQuickPickItem("Top level", TemplateSectionType.TopLevel, "Sort top level items based on recommended order (parameters, functions, variables, resources, outputs)"));
     return items;
 }
 
-export async function sortTemplate(template: DeploymentTemplate | undefined, sortType: SortType, textEditor: vscode.TextEditor): Promise<void> {
+export async function sortTemplate(template: DeploymentTemplate | undefined, sectionType: TemplateSectionType, textEditor: vscode.TextEditor): Promise<void> {
     if (!template) {
         return;
     }
     ext.outputChannel.appendLine("Sorting template");
-    switch (sortType) {
-        case SortType.Functions:
+    switch (sectionType) {
+        case TemplateSectionType.Functions:
             await sortFunctions(template, textEditor);
             break;
-        case SortType.Outputs:
+        case TemplateSectionType.Outputs:
             await sortOutputs(template, textEditor);
             break;
-        case SortType.Parameters:
+        case TemplateSectionType.Parameters:
             await sortParameters(template, textEditor);
             break;
-        case SortType.Resources:
+        case TemplateSectionType.Resources:
             await sortResources(template, textEditor);
             break;
-        case SortType.Variables:
+        case TemplateSectionType.Variables:
             await sortVariables(template, textEditor);
             break;
-        case SortType.TopLevel:
+        case TemplateSectionType.TopLevel:
             await sortTopLevel(template, textEditor);
             break;
         default:
```

Comment:
case TemplateSectionType.TopLevel:
                assert.fail("Unknown insert item type!");
            default:
                assertNever(sectionType);

Reply:
assertNever will give a compile error (not runtime) if there's a missed case.

Reply:
Fixed.